### PR TITLE
Fix some misprints and mistakes in code

### PR DESCRIPTION
**Pull request by UnrealKaraulov**
Fix some critical errors in AMXMODX!

### CODE REVIEW

**Comment by UnrealKaraulov** [comment]
@ShootingKing-AM sorry for ping :(

But you can merge it ? (fix some buffers overflows, fix misprints)

File: amxmodx/amx.cpp [hunk 1]
```diff
@@ -1416,7 +1416,7 @@ static AMX_NATIVE findfunction(const char *name, const AMX_NATIVE_INFO *list, in
   int i;
 
   assert(list!=NULL);
-  for (i=0; list[i].name!=NULL && (i<number || number==-1); i++)
```
Comment (dvander) [inline amxmodx/amx.cpp:1419]: These lists are null terminated. When is number == -1? Is this change from upstream?

File: amxmodx/file.cpp [hunk 2]
```diff
@@ -96,12 +96,14 @@ static cell AMX_NATIVE_CALL read_file(AMX *amx, cell *params)
 	{
 		length = strlen(buffer);
 
-		if (buffer[length - 1] == '\n')
-			buffer[--length] = '\0';
-
-		if (buffer[length - 1] == '\r')
-			buffer[--length] = '\0';
-
+        if (length > 0)
```
Comment (dvander) [inline amxmodx/file.cpp:99]: Indentation is wrong here - looks like your editor is not inserting hard tabs.

File: compiler/libpc300/sc2.c [hunk 3]
```diff
@@ -2950,7 +2950,7 @@ static char itohstr[30];
     val>>=4;
   } /* endfor */
   i=max-1;
-  while (nibble[i]==0 && i>0)   /* search for highest non-zero nibble */
+  while (i>0 && nibble[i]==0)   /* search for highest non-zero nibble */
```
Comment (dvander) [inline compiler/libpc300/sc2.c:2953]: What is the bug here?

File: modules/cstrike/csx/WinCSX/WinCSX.cpp [hunk 4]
```diff
@@ -229,8 +229,8 @@ void ListboxItemSelected(HWND hDlg) {
 	// Retrieve complete stats record of this position. Position in listbox should be same as rank in our records!
 	RankSystem::RankStats* stats = g_rank.findEntryInRankByPos((int)nItem + 1);
 	if (stats == NULL) {
-		char msg[] = "Error: Couldn't find the record by position! (nItem = %d)";
-		sprintf(msg, msg, nItem);
+		char msg[512];
+		sprintf(msg, "Error: Couldn't find the record by position! (nItem = %d)", nItem);
```
Comment (dvander) [inline modules/cstrike/csx/WinCSX/WinCSX.cpp:233]: Pretty clearly a bug being fixed here, but it's sad that AMXX is pretty old and uses sprintf() everywhere. It should be ke::SafeSprintf (or snprintf).

File: modules/ts/tsx/NRank.cpp [hunk 5]
```diff
@@ -295,10 +295,10 @@ static cell AMX_NATIVE_CALL cwpn_dmg(AMX *amx, cell *params){ // wid,att,vic,dmg
 	CPlayer* pAtt = GET_PLAYER_POINTER_I(att);
 	CPlayer* pVic = GET_PLAYER_POINTER_I(vic);
 
+    if ( !pAtt ) pAtt = pVic;
```
Comment (dvander) [inline modules/ts/tsx/NRank.cpp:298]: Bad indentation again.

File: modules/ts/tsx/usermsg.cpp [hunk 6]
```diff
@@ -122,7 +122,8 @@ void Client_TSHealth_End(void* mValue){
 			
 			weapon = 37; // throwing knife
 			aim = pAttacker ? pAttacker->aiming : 0;
-			pAttacker->saveHit( mPlayer , weapon , damage, aim );
+            if (pAttacker)
```
Comment (dvander) [inline modules/ts/tsx/usermsg.cpp:125]: Mode indentation problems.

File: compiler/libpc300/sc2.c (hunk 3 above)
Comment (UnrealKaraulov) [reply compiler/libpc300/sc2.c:2953]: i can be -1 ?

File: modules/cstrike/csx/WinCSX/WinCSX.cpp (hunk 4 above)
Comment (UnrealKaraulov) [reply modules/cstrike/csx/WinCSX/WinCSX.cpp:233]: yes

File: compiler/libpc300/sc2.c (hunk 3 above)
Comment (dvander) [reply compiler/libpc300/sc2.c:2953]: I don't see how. `i` starts at 7, and when it reaches 0, the condition will fail.

File: amxmodx/file.cpp (hunk 2 above)
Comment (UnrealKaraulov) [reply amxmodx/file.cpp]: Yes sorry

**Comment by UnrealKaraulov** [comment]
@dvander thanks, fixed problems.

**Comment by dvander** [comment]
Thanks! I'll merge this tomorrow morning.

**Pull request merged**